### PR TITLE
Unit Tests No Longer Launch App Fully

### LINKED ISSUE #7046: Unit Testing: Don't Load WordPressAppDelegate
Simplify the unit testing launch by not loading the main app delegate when launching. This should reduce complexities with testing code without a Core Data stack and other things loaded in memory.

I'm sure this will also cause issues with the "shared instance" property on the App Delegate and may be a rabbit hole.

### PR DESCRIPTION
**Pull request by astralbodies**
Closes #7046 

This PR disconnects the WordPressAppDelegate when running the unit tests. This means that none of the app is really brought up into a running state (Analytics, Core Data, Reachability, etc). It results in both a quicker launch time for tests and also totally breaks any dependencies tests have on the state of the app as a whole.

The TestingAppDelegate class replaces WordPressAppDelegate inside of main.m when unit tests are running (specifically when the test bundle is injected and that class exists). The old main UI XIBs were deleted and replaced with an instantiation of the main UIWindow in the regular app delegate.

## Testing
1. Launch the app as usual in both iPhone & iPad simulators/devices.
2. Make sure rotations work as expected, backgrounding, multitasking, etc. Just want to ensure the UIWindow creation isn't missing any settings that the XIBs may have set.
3. Kill the app and launch the unit tests. Make sure they all pass.

## Things to Note
I'm seeing a bunch of constraint break warnings in the console but I'm not 100% sure this PR didn't introduce them. @frosty & @kurzee you may want to take a look as I've seen split VC mentioned in the warnings.

@jleandroperez I had to update the notifications manager test to bring up an instance of Helpshift to test the class. Make sure I didn't break anything or am giving the test false positive results. 😄 

Needs review: @frosty, @kurzee, @jleandroperez, @koke 

### Special Thanks
@jonreid and his [App Delegate testing post](http://qualitycoding.org/app-delegate-for-tests/).

### CODE REVIEW

**Comment by SergioEstevao** [comment]
@astralbodies really liked the way you implemented this!

File: WordPress/Classes/System/WordPressAppDelegate.h [hunk 1]
```diff
@@ -11,13 +11,13 @@
...
-@property (nonatomic, strong,  readonly) Reachability                   *wpcomReachability;
-@property (nonatomic, assign,  readonly) BOOL                           connectionAvailable;
-@property (nonatomic, strong,  readonly) WPUserAgent                    *userAgent;
+@property (nonatomic, strong, readonly) WPAppAnalytics *analytics;
+@property (strong, nonatomic) UIWindow *window;
+@property (nonatomic, strong, readonly) WPLogger *logger;
+@property (nonatomic, strong, readonly) Reachability *internetReachability;
+@property (nonatomic, strong, readonly) Reachability *wpcomReachability;
+@property (nonatomic, assign, readonly) BOOL connectionAvailable;
+@property (nonatomic, strong, readonly) WPUserAgent *userAgent;
```
Comment (jleandroperez) [inline WordPress/Classes/System/WordPressAppDelegate.h]: Note to self: Okay. Relax. Breathe. It's just lack of alignment.

File: WordPress/Classes/System/WordPressAppDelegate.m [hunk 2]
```diff
@@ -120,7 +122,6 @@ - (BOOL)application:(UIApplication *)application willFinishLaunchingWithOptions:
 - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions
 {
     DDLogVerbose(@"didFinishLaunchingWithOptions state: %d", application.applicationState);
-    [self.window makeKeyAndVisible];
```
Comment (jleandroperez) [inline WordPress/Classes/System/WordPressAppDelegate.m]: Out of curiosity: Is this actually **not** needed? instantiating a UIWindow makes it Key and Visible by default?

File: WordPress/Classes/System/WordPressAppDelegate.h (hunk 1 above)
Comment (catehstn) [inline WordPress/Classes/System/WordPressAppDelegate.h]: 😱

File: WordPress/Classes/System/WordPressAppDelegate.m (hunk 2 above)
Comment (astralbodies) [inline WordPress/Classes/System/WordPressAppDelegate.m]: @jleandroperez We're calling the same method a bit later in this method.

**Comment by astralbodies** [comment]
@koke

**Comment by bummytime** [comment]
Nice work @astralbodies ❗️🥇

**Comment by nheagy** [comment]
I'm really excited to see the tests run faster!

**Pull request merged**